### PR TITLE
chore: add mixpanel batch size metrics

### PR DESCRIPTION
**Pull request by Gauravudia**
## What are the changes introduced in this PR?

Adding batch size for each endpoint of `Mixpanel` destination
Screenshot 2024-03-21 at 4 41 22 PM

## What is the related Linear task?

Resolves INT-1878

## Please explain the objectives of your changes below

We are capturing this to get the info about how much each endpoint is packing.

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [x] Is the type of change in the PR title appropriate as per the changes?

- [x] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3201/test-report.html

**Comment by yashasvibajpai** [comment]
@Gauravudia can you update the PR template with some details on why batch size info is needed for all different endpoints being hit?

**Comment by Gauravudia** [comment]
> @Gauravudia can you update the PR template with some details on why batch size info is needed for all different endpoints being hit?

We are capturing this to get the info about how much each endpoint is packing.

File: src/v0/destinations/mp/transform.js [hunk 1]
```diff
@@ -510,6 +516,11 @@ const processRouterDest = async (inputs, reqMetadata) => {
       const { engageEvents, groupsEvents, trackEvents, importEvents, batchErrorRespList } =
         groupEventsByEndpoint(transformedPayloads);
 
+      engageBatchSize += engageEvents.length;
+      groupsBatchSize += groupsEvents.length;
+      trackBatchSize += trackEvents.length;
+      importBatchSize += importEvents.length;
+
       const engageRespList = batchEvents(engageEvents, ENGAGE_MAX_BATCH_SIZE, reqMetadata);
```
[the remark this answers is not in the document]
Comment (Gauravudia) [reply src/v0/destinations/mp/transform.js]: Addressed

**Review by utsabc — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**